src/Components/Analyzers/src/ComponentInternalUsageDiagnosticAnalzyer.cs

```diff
@@ -27,17 +30,25 @@ public ComponentInternalUsageDiagnosticAnalyzer()
 
         public override void Initialize(AnalysisContext context)
         {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
             _inner.Register(context);
         }
 
         private static bool IsInInternalNamespace(ISymbol symbol)
         {
-            if (symbol?.ContainingNamespace?.ToDisplayString() is string ns)
+            var @namespace = symbol?.ContainingNamespace;
+            for (var i = 0; i < NamespaceParts.Length; i++)
             {
-                return string.Equals(ns, "Microsoft.AspNetCore.Components.RenderTree");
+                if (@namespace == null || !string.Equals(NamespaceParts[i], @namespace.Name, StringComparison.Ordinal))
+                {
+                    return false;
+                }
+
+                @namespace = @namespace.ContainingNamespace;
             }
 
-            return false;
+            return true;
```

Comment:
This was the optimization in this change. Comparing these by ToString is slow, and bad (I'm told).

Reply:
```suggestion
            return @namespace.IsGlobalNamespace;
```

Reply:
To avoid weird things like 'My.Microsoft.AspNetCore.Components.RenderTree`